source/loader/layers/sanitizer/linux/symbolizer.cpp

```diff
@@ -12,9 +12,21 @@
 
 namespace ur_sanitizer_layer {
 
+static llvm::symbolize::LLVMSymbolizer *Symbolizer = nullptr;
+
 llvm::symbolize::LLVMSymbolizer *GetSymbolizer() {
```

Comment:
can't you simply hold this value inside of the layer's context? Those structures are carefully managed to avoid lifetime issues. I don't like having related globals if we can avoid them.

Reply:
That would be another option. 

But I want to avoid having such value inside the layer's context because I think the layer does not have to know about this symbolizer and that makes them coupled. And this will only construct the symbolizer when needed.

Also, according to our teammates, we are having this "linux" folder because they may not be available for other platforms(although I don't think we will support other platforms in near future), and that is another thing. 

And having a symbolizer is not also true for such case because the symbolizer is not available by default. 

@zhaomaosu suggests that we can expose a weak function that destructs the symbolizer and calls it in the destructor of SanitizerInterceptor.

Reply:
I know this makes the code less clean, but I think it's a compromise worth making. The constructor/destructor ordering in sycl rt is very very fragile, and 
I guarantee that relaying on existing mechanism (the layer contexts) will be less error prone long term.

Reply:
Updated already, please review, thanks.